lcls-twincat-math/lcls_twincat_math/POUs/FB_Vec3.TcPOU

```diff
@@ -0,0 +1,289 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<TcPlcObject Version="1.1.0.1">
+  <POU Name="FB_Vec3" Id="{c09dfe87-9fa9-4800-966a-c9fa2f01d9b6}" SpecialFunc="None">
+    <Declaration><![CDATA[
+FUNCTION_BLOCK FB_Vec3 IMPLEMENTS I_Vec3, I_Quat
```

Comment:
Why does FB_Vec3 implement I_Quat? Naiively I'd assume it would only implement I_Vec3

Reply:
@ZLLentz because when using quaternion math to manipulate 3D vectors, you first need to convert the vector into a quaternion. By having a Vec3 implement the quaternion interface, it allows math between quaternions and Vec3s to be a more convenient to do. E.g, to multiply a vec3 by a quaternion I don't have to always allocate a new FB_Quaternion just to do a few operations and then convert back to a Vec3. And the conversion of a Vec3 to a quat is well-defined, just set w to zero and use x,y,z for i,j,k mutipliers of the quaternion.